File: LLD-Python/solid_principle/dependency_inversion_principle/dependency_inversion_bad.py
class EmailSender:
    def send(self, message: str) -> None:
        # send email
        pass

class NotificationService:
    def __init__(self)-> None:
        self.email_sender = EmailSender()

    def notify(self, message: str) -> None:
        self.email_sender.send(message)

File: LLD - Java/src/main/java/org/solid/dependencyInversionPrinciple/DependencyInversionBad.java
package org.solid.dependencyInversionPrinciple;

public class DependencyInversionBad {
    
}

class EmailSender {
    void send(String message) {/* send mail */}
}

class NotificationService {
    EmailSender emailSender = new EmailSender();

    void notify(String message) {
        emailSender.send(message);
    }
}